pyamg/aggregation/adaptive.py

```diff
@@ -113,7 +113,7 @@ def unpack_arg(v):
 def adaptive_sa_solver(A, initial_candidates=None, symmetry='hermitian',
                        pdef=True, num_candidates=1, candidate_iters=5,
                        improvement_iters=0, epsilon=0.1,
-                       max_levels=10, max_coarse=100, aggregate='standard',
```

Comment:
I did some more thinking, and I'm not sure that the pinv() FLOPs should be counted in the CC.  They're just such a different type of FLOP than sparse matrix FLOPs that it seems like we're counting apples and oranges together.  LAPACK gets a great deal of memory re-use through things like cache blocking and speedups from tweaks like loop unrolling.

From the perspective of a scaling study, all that we'd be doing is ignoring a fixed cost when we compute the CC.  We'd still see the same trend line of scalability (or lack thereof).  Also, including an n^3 term in the CC could skew the total work units.  The relative cost in work units of the coarsest-grid solve would decrease as you go to larger problem sizes, possibly making the CC smaller for larger grid sizes.

Hypre does have a small default max_coarse_size of around 10, but what people use in parallel varies quite a bit.  There's a benefit to removing the last few levels in a hierarchy, in that those levels have dense rows that induce a lot of communication.  Marian used to say he stopped at a coarse-level size of about 10,000 and then called a parallel direct sparse solver.  

….At the end of it all, I don’t have a strong preference on the default max_coarse.  I’d probably choose something like 100, but 10 is probably OK too.  But, I do lean towards not including the coarsest grid matrix inversion in the CC, especially if its fixed and relatively small.